countdown.js

```diff
@@ -0,0 +1,42 @@
+//console.log("contdown.js is running");
+
+//Here is a simple version of the countdown taken from https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_countdown
+// Set the date we're counting down to
+
+//const countDownDate = new Date("Nov 13, 2020 00:00:00").getTime();
+//use this line to have countdown for 10 seconds only - for testing purpose only
+const countDownDate = new Date().getTime() + 10000; 
+
+const mainDivElem = document.querySelector("#mainDiv");
+mainDivElem.style.display = "none";
+
+// Update the count down every 1 second
+let x = setInterval(function() {
+
+  // Get today's date and time
+  const now = new Date().getTime();
+    
+  // Find the distance between now and the count down date
+  const distance = countDownDate - now;
+
+    
+  // Time calculations for days, hours, minutes and seconds
+  const days = Math.floor(distance / (1000 * 60 * 60 * 24));
+  const hours = Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  const minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
+  const seconds = Math.floor((distance % (1000 * 60)) / 1000);
+
+  const countdownDivElem = document.querySelector("#countdownDiv");
+  // Output the result in an element with id="demo"
+  countdownDivElem.innerHTML =
+  `<i class="fas fa-gift"></i>
+   ${days}d ${hours}h ${minutes}m ${seconds}s
+   <i class="fas fa-gift"></i>`;
+  // If the count down is over, write some text 
```

Comment:
From a previous class, I kind of got the idea that we shouldn't add html elements like strings in js "<i class="fas fa-gift"></i>", in here that would involve making a span element to display the count down, and an extra id to make the div disappear.  
<div id="countdownDiv"> <i class="fas fa-gift"></i><span id= "countdownSpan" ><span><i class="fas fa-gift">
</i></div>
It would be longer but as I understand divs and spans are like commits you can use as many as you need. It is up to you.

Reply:
You're right, during the class Sebastian was talking about that. But he meant, that when you want to ADD one more child element to an already existing parent element it's not a good practice to ADD a string, containing this element to the NOT EMPTY `parent.innerHTML` string. Here I do not add the content, I assign the whole `innerHTML` completely. To do like that is ok.
Sebastian has this kind of code [here](https://github.com/marsamak/redi-js-code-example-2020-10-20/blob/master/fetch/fetch.js) as well.